framework/src/kernels/ADKernel.C

```diff
@@ -167,12 +167,24 @@ template <typename T, ComputeStage compute_stage>
 void
 ADKernelTempl<T, compute_stage>::computeADOffDiagJacobian()
 {
-  std::vector<DualReal> residuals(_test.size(), 0);
+  if (_residuals.size() != _test.size())
+    _residuals.resize(_test.size(), 0);
+  for (auto & r : _residuals)
+    r = 0;
 
   precalculateResidual();
+  ADResidual tmp;
   for (_qp = 0; _qp < _qrule->n_points(); _qp++)
```

Comment:
Wait - why are you adding your own temporary?  Make this part of the class so it doesn’t get created and destroyed each time.

Reply:
fixed